### PR TITLE
Reduce prefetch count on restart and gradually restore it

### PR DESCRIPTION
**Pull request by thedrow**
*Note*: Before submitting this pull request, please review our [contributing
guidelines](https://docs.celeryproject.org/en/master/contributing.html).

### CODE REVIEW

**Comment by dobosevych** [comment]
@auvipy there was an issue with Mock * int that I fixed in this PR

**Comment by thedrow** [comment]
You're right. It needs to be documented.

**Review by matusvalo — APPROVED** [review]
LGTM

**Review by auvipy — CHANGES_REQUESTED** [review]
we actually need this documented.

**Comment by thedrow** [comment]
Yes. I'm adding more logs now and this feature will be mentioned in the user guide

File: docs/userguide/workers.rst [hunk 1]
```diff
@@ -137,6 +137,22 @@ isn't recommended in production:
     :sig:`HUP` is disabled on macOS because of a limitation on
     that platform.
 
+Automatic re-connection on connection loss to broker
+====================================================
```
Comment (auvipy) [inline docs/userguide/workers.rst:141]: i think just version added 5.3 notation needed only here

**Comment by auvipy** [comment]
thanks!

**Pull request merged**